### DISCUSSION TITLE
A function to replace the select part for one column

### DISCUSSION THREAD

**Discussion by mohs8421** [SeaQL/sea-orm#1644] [Ideas]
## Motivation
When I have a larger struct, from which I want to have most columns as is, but let's say I want to have one of the existing columns with a function call like upper case, or COALESCE(), then I am forced to call `.select_only()` and repeat all the other columns.

## Proposed Solutions
So I would like to have a function, which allows me to write the desired select piece, without being forced to do anything on the other columns.
This could be achieved by having all columns in the selection being mapped by the alias, and you could replace the selection for that alias with something new.
e.g.:
```rust
  fn column_instead<S: Into<SimpleExpr>>(mut self, expression: S, alias: &str) -> Self {
    …
  }
```

**Comment by billy1624** [comment]
Hey @mohs8421, interesting!! I have a counter proposal!

How about we allow user to supply the list of selected column at the very beginning?

```rs
let cols = cake::Column::iter()
    .map(|col| {
        let expr = col.into_simple_expr();
        match col {
            cake::Column::Name => Func::upper(expr).into(),
            _ => expr,
        }
    });

cake::Entity::select(cols)
    .all(db)
    .await?;
```

**Comment by mohs8421** [comment]
that way you would still need to know more about the way sea-orm works internally, if you reduce your proposal to not use map with a callback, but with a filter, that would be easier, or is it possible to offer both by having the select definition like
```rust
fn select<S: IntoSimpleExpr>(cols: Vec<S>)…
```

**Comment by mohs8421** [comment]
another thought regarding this: If you join a foreign table, you might want to have some columns of that table with a common prefix, so if you could stack the select() calls, or select_prefix() calls, that would cover these use cases as well.
